src/Components/Trainers/Form/index.jsx

```diff
@@ -34,114 +64,206 @@ const Form = ({ add, edit, onUpdate }) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainer/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(updatedTrainer)
+      });
+      if (response.ok) {
+        setResponseMessage('Trainer updated successfully');
+        setShowResponseModal(true);
+        setStateModal('success');
+        setTimeout(() => {
+          setShowResponseModal(false);
+          history.push('/trainers');
+        }, 1500);
+      } else {
+        setResponseMessage('Failed to update trainer');
+        setShowResponseModal(true);
+        setStateModal('fail');
+        setTimeout(() => {
+          setShowResponseModal(false);
+        }, 3000);
+      }
+    } catch (error) {
+      setResponseMessage(`Error updating trainer: ${error.message}`);
+      setShowResponseModal(true);
+      setStateModal('fail');
+    }
+  };
+
+  const handleConfirmModal = () => {
+    if (validateInputs()) {
+      setShowConfirmModal(true);
     } else {
-      add(trainer);
+      setResponseMessage('Please fill in all fields');
+      setShowResponseModal(true);
+      setStateModal('fail');
+      setTimeout(() => {
+        setShowResponseModal(false);
+      }, 3000);
     }
   };
 
+  const validateInputs = () => {
+    for (const key in trainer) {
+      if (trainer[key].trim() === '') {
+        return false;
+      }
+    }
+    return true;
+  };
+
   return (
-    <form onSubmit={onSubmit} className={styles.container}>
-      <div className={styles.flex}>
-        <label htmlFor="firstName" className={styles.label}>
-          First Name
-        </label>
-        <input
-          name="firstName"
-          type="text"
-          id="firstName"
-          value={trainer.firstName}
-          onChange={onChangeInput}
-          className={styles.input}
-        />
-      </div>
-      <div className={styles.flex}>
-        <label htmlFor="lastName" className={styles.label}>
-          Last Name
-        </label>
-        <input
-          name="lastName"
-          type="text"
-          id="lastName"
-          value={trainer.lastName}
-          onChange={onChangeInput}
-          className={styles.input}
-        />
-      </div>
-      <div className={styles.flex}>
-        <label htmlFor="dni" className={styles.label}>
-          ID
-        </label>
-        <input
-          name="dni"
-          type="text"
-          id="dni"
-          value={trainer.dni}
-          onChange={onChangeInput}
-          className={styles.input}
-        />
+    <>
+      <h2 className={styles.title}>{id ? 'Edit Trainer' : 'Add Trainer'}</h2>
+      <form className={styles.container}>
+        <div className={styles.flex}>
+          <Input
+            labelText="First Name"
+            name="firstName"
+            type="text"
+            value={trainer.firstName}
+            change={onChangeInput}
+            className={styles.input}
```

Comment:
hi mauro, here you dont need to pass a className in the input component because it dont receive a prop named "className"

Reply:
and that apply to all Inputs